tests/src/Kernel/Controller/WopiControllerTest.php

```diff
@@ -0,0 +1,432 @@
+
+/**
+ * @coversDefaultClass \Drupal\collabora_online\Controller\WopiController
+ */
+class WopiControllerTest extends CollaboraKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'collabora_online_test',
+  ];
+
+  /**
+   * The user with access to perform operations.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected UserInterface $user;
+
+  /**
+   * The media where to perform operations.
+   *
+   * @var \Drupal\media\MediaInterface
+   */
+  protected MediaInterface $media;
+
+  /**
+   * The source file.
+   *
+   * @var \Drupal\file\FileInterface
+   */
+  protected FileInterface $file;
+
+  /**
+   * The test logger channel.
+   *
+   * @var \ColinODell\PsrTestLogger\TestLogger
+   */
+  protected TestLogger $logger;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->logger = new TestLogger();
+    \Drupal::service('logger.factory')->addLogger($this->logger);
+
+    $collabora_settings = \Drupal::configFactory()->getEditable('collabora_online.settings');
+    $cool = $collabora_settings->get('cool');
+    $cool['key_id'] = 'collabora';
+    $collabora_settings->set('cool', $cool)->save();
+
+    // Make sure that ids for different entity types are distinguishable.
+    // This will reveal bugs where one id gets mixed up for another.
+    \Drupal::database()->query("ALTER TABLE {media} AUTO_INCREMENT = 1000");
+    \Drupal::database()->query("ALTER TABLE {file_managed} AUTO_INCREMENT = 2000");
+
+    $this->media = $this->createMediaEntity('document');
+    $this->user = $this->createUser([
+      'access content',
+      'edit any document in collabora',
+    ]);
+    $fid = $this->media->getSource()->getSourceFieldValue($this->media);
+    $this->file = File::load($fid);
+
+    $this->setCurrentUser($this->user);
+  }
+
+  /**
+   * Tests successful requests to the 'collabora-online.wopi.info' route.
+   *
+   * @covers ::wopiCheckFileInfo
+   */
+  public function testWopiGetFileInfo(): void {
+    $file_changed_time = date_create_immutable_from_format('U', (string) $this->file->getChangedTime());
+    $assert_response = function (Request $request, bool $write) use ($file_changed_time) {
+      $this->assertResponse(
+        Response::HTTP_OK,
+        json_encode([
+          'BaseFileName' => $this->file->getFilename(),
+          'Size' => $this->file->getSize(),
+          'LastModifiedTime' => $file_changed_time->format('c'),
+          'UserId' => $this->user->id(),
+          'UserFriendlyName' => $this->user->getDisplayName(),
+          'UserExtraInfo' => [
+            'mail' => $this->user->getEmail(),
+          ],
+          'UserCanWrite' => $write,
+          'IsAdminUser' => FALSE,
+          'IsAnonymousUser' => FALSE,
+        ]),
+        'application/json',
+        $request,
+      );
+    };
+
+    $request = $this->createRequest();
+    $assert_response($request, FALSE);
```

Comment:
Can we take out the assertion assertResponse() from inside the inline function and just have the inline function generate the expected response?
It's more clear for reading, and we keep the failure in this line instead of having to look at the call stack.